common/include/pcl/common/impl/spring.hpp

```diff
@@ -80,7 +80,7 @@ expandColumns (const PointCloud<PointT>& input, PointCloud<PointT>& output,
   output.height = old_height;
 }
 
-template <typename PointT, typename T = pcl::index_t, std::enable_if_t<!std::is_same<T, std::size_t>::value, pcl::index_t> = 0> void
+template <typename PointT, typename T, std::enable_if_t<!std::is_same<T, std::size_t>::value, pcl::index_t>> void
```

Comment:
If you actually don't need the type here, you probably can write it simply as 
```cpp
template <typename PointT, typename, pcl::index_t> void
```

give it a try. The same is applicable to all other changes in this commit.

Reply:
SFINAE is needed. I'd recommend using a standin and comparing `T` with both `index_t` and whatever type `index_t` is replacing:

```cpp
template <class T, class OldType>
using enable_for_compatibility = std::enable_if_t<std::is_same_v<T, OldType> && !std::is_same_v<T, index_t>, bool>;

template <class Point, typename T, enable_for_compatibility<T, int> = 0>
void foo(T bar);
```

https://godbolt.org/z/zPede6

Reply:
If it's still needed then leave it untouched.

Reply:
@haritha-j Please take a look at the SFINAE (and my sample) again. You've technically created a lot of overloads instead of 1

Reply:
The reason I removed the type here was because one of the CI pipelines (Windows I think) fails when the type is assigned in the header file and then assigned again in the `hpp` file. The `.h` file still looks like this:
```cpp 
    template <typename PointT, typename T = pcl::index_t, std::enable_if_t<!std::is_same<T, std::size_t>::value, pcl::index_t> = 0>
```
@kunaltyagi do I still need to use the method in your example?

Reply:
> @haritha-j Please take a look at the SFINAE (and my sample) again. You've technically created a lot of overloads instead of 1

I'm not sure I agree with this. I've made the following test https://godbolt.org/z/8rc5jd
I've appended to your example the old SFINAE pattern and I instantiated the function with 4 different integer types.  Only two overloads exist as intended.

Reply:
That seems like a very unlikely scenario IMO. I don't really see any reason someone would attempt to manually set `T`.

Reply:
If this issue crops up, we can fix it without breaking anything (or just remove it in 2 releases)